packages/blade/src/components/Modal/Modal.web.tsx

```diff
@@ -0,0 +1,245 @@
+  initialFocusRef?: React.MutableRefObject<any>;
+  /**
+   *  Size of the modal
+   * @default 'small'
+   */
+  size?: 'small' | 'medium' | 'large';
+  /**
+   *  Accessibility label for the modal
+   */
+  accessibilityLabel?: string;
+};
+
+const entry = keyframes`
+  from {
+    opacity: 0;
+    transform: translate(-50%, -50%) scale(0.9) translateY(20px);
+  }
+  to {
+    opacity: 1;
+    transform: translate(-50%, -50%) scale(1) translateY(0px);
+  }
+`;
+
+const exit = keyframes`
+  from {
+    opacity: 1;
+    transform: translate(-50%, -50%) scale(1) translateY(0px);
+  }
+  to {
+    opacity: 0;
+    transform: translate(-50%, -50%) scale(0.9) translateY(20px);
+  }
+`;
+
+const ModalContent = styled(BaseBox)<{ isVisible: boolean }>(({ isVisible, theme }) => {
+  return css`
+    box-shadow: ${theme.elevation.highRaised};
+    position: fixed;
+    transform: translate(-50%, -50%);
+    opacity: ${isVisible ? 1 : 0};
+    animation: ${isVisible ? entry : exit}
+      ${castWebType(makeMotionTime(theme.motion.duration.xmoderate))}
+      ${isVisible
+        ? castWebType(theme.motion.easing.entrance.revealing)
+        : castWebType(theme.motion.easing.exit.revealing)};
+  `;
+});
+
+const Modal = ({
+  isOpen = false,
+  children,
+  onDismiss,
+  initialFocusRef,
+  size = 'small',
+  accessibilityLabel,
+}: ModalProps): React.ReactElement => {
+  const { theme, platform } = useTheme();
+  const [footerHeight, setFooterHeight] = useState(0);
+  const { isMounted, isVisible } = usePresence(isOpen, {
+    transitionDuration: theme.motion.duration.xmoderate,
+    initialEnter: true,
+  });
+
+  // Warn consumer if modal is opened on mobile
+  useEffect(() => {
+    if (platform === 'onMobile') {
+      console.warn(
+        '[Blade Modal] Modal is not supported on mobile devices. Please use BottomSheet instead.',
+      );
+    }
+  }, [platform]);
+
+  // required by floating ui to handle focus
+  const { refs, context } = useFloating({
+    open: isMounted,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const defaultInitialFocusRef = React.useRef<any>(null);
+  const originalFocusElement = React.useRef<HTMLElement | null>(null);
+
+  const returnFocus = React.useCallback(() => {
+    if (!originalFocusElement.current) return;
+    originalFocusElement.current.focus();
+    // After returning focus we will clear the original focus
+    // Because if modal can be opened up via multiple triggers
+    // We want to ensure the focus returns back to the most recent triggerer
+    originalFocusElement.current = null;
+  }, [originalFocusElement]);
+
+  const focusOnInitialRef = React.useCallback(() => {
+    if (!initialFocusRef) {
+      // focus on close button
+      defaultInitialFocusRef.current?.focus();
+    } else {
+      // focus on the initialRef passed by the user
+      initialFocusRef.current?.focus();
+    }
+  }, [initialFocusRef]);
```

Comment:
All these logic probably can be removed. 

FloatingFocusManager supports focus return. 

```
<FloatingFocusManager
  initialFocus={initialFocusRef || defaultInitialFocusRef}
  returnFocus
  context={context}
  modal={true}
>
```

Reply:
Weird, could be some sort of timing issue. But anyways we can pick up a refactor task later. [Tracking it here](https://github.com/razorpay/blade/issues/1269)

Reply:
Not related to timing, I think it might be related to how it manages focus. I can see my element nested within the element its adding focus to.